torch/_inductor/lowering.py

```diff
@@ -3895,6 +3896,44 @@ def _realize(x):
     return clone(x)
 
 
+# The difference between quantize_per_tensor.default and quantize_per_tensor.tensor is
+# scale and zero_point is scalar or scalar tensor
+@register_lowering(torch.ops.quantized_decomposed.quantize_per_tensor.default)
+def quantize_per_tensor_default(input, scale, zero_point, quant_min, quant_max, dtype):
+    def clamp(v):
+        return maximum(quant_min, minimum(quant_max, v))
+
+    output = clamp(add(round(div(input, scale)), zero_point))
+    output = to_dtype(output, dtype)
+    return output
+
+
+# The difference between dequantize_per_tensor.default and dequantize_per_tensor.tensor is
+# scale and zero_point is scalar or scalar tensor
+@register_lowering(torch.ops.quantized_decomposed.dequantize_per_tensor.default)
+def dequantize_per_tensor_default(
+    input, scale, zero_point, quant_min, quant_max, dtype
+):
+    output = mul(sub(to_dtype(input, torch.float32), zero_point), scale)
+    return output
+
+
+@register_lowering(torch.ops.quantized_decomposed.quantize_per_tensor.tensor)
+def quantize_per_tensor_tensor(input, scale, zero_point, quant_min, quant_max, dtype):
+    def clamp(v):
+        return maximum(quant_min, minimum(quant_max, v))
+
+    output = clamp(add(round(div(input, scale)), zero_point))
+    output = to_dtype(output, dtype)
+    return output
+
+
+@register_lowering(torch.ops.quantized_decomposed.dequantize_per_tensor.tensor)
+def dequantize_per_tensor_tensor(input, scale, zero_point, quant_min, quant_max, dtype):
+    output = mul(sub(to_dtype(input, torch.float32), zero_point), scale)
+    return output
+
+
```

Comment:
These might be cleaner to write as decomps in `decomposition.py`

Reply:
Hi @jansel, thanks for your comments. There is already an existing but un-landed PR https://github.com/pytorch/pytorch/pull/97321 from quantization team to register these operations into `decomposition table`. However, synced with @jerryzh168 previously, quantization team has not decided whether to decompose these operations yet. That's why we do these operations lowering here. Do you think it's ok we do this lowering firstly and remove them after quantization team decide to merge above PR https://github.com/pytorch/pytorch/pull/97321? also cc @jgong5 @EikanWang here.

Reply:
If we decide not to merge #97321, you can just put the decomps in the inductor codebase here: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/decomposition.py

When you call AotAutograd you get to pick your own decomps, so even if inductor decomposes it others can decide not to.

The above code does the exact same thing as using decomps, just in a slightly hard to read way.

Reply:
@jansel Thanks for your suggestions. Move these ops into https://github.com/pytorch/pytorch/blob/main/torch/_inductor/decomposition.py. Please help to take a look again.